components/table/ReportTable.tsx

```diff
@@ -26,20 +28,52 @@ interface ProjectHoursRowProps {
 interface EmployeeHoursRowProps {
     entries: TimesheetEntry[]
     employee: Employee
+    projects: Project[]
 }
 
 interface ReportTableProps {
     entries: TimesheetEntry[]
     customers: Customer[]
     projects: Project[]
     employees: Employee[]
+    timesheets: Timesheet[]
     startDate: string
     endDate: string
 }
 
 const entriesQuantitySum = (entries: TimesheetEntry[]) =>
     entries.reduce((total, currentItem) => total + currentItem.quantity, 0)
 
+const entriesByProject = (
+    entriesToFilter: TimesheetEntry[],
+    projectId: number
+) =>
+    entriesToFilter.filter((entry) => entry.timesheet.project?.id === projectId)
+
+const entriesByCustomer = (
+    entriesToFilter: TimesheetEntry[],
+    customerId: number
+) =>
+    entriesToFilter.filter(
+        (entry) => entry.timesheet.project.customer?.id === customerId
+    )
+const entriesByEmployee = (
+    entriesToFilter: TimesheetEntry[],
+    employeeId: number
+) =>
+    entriesToFilter.filter(
+        (entry) => entry.timesheet.employee?.id === employeeId
+    )
+const projectByCustomer = (projectsToFilter: Project[], customerId: number) =>
+    projectsToFilter.filter((project) => project.customer?.id === customerId)
+const timesheetsByEmployee = (
```

Comment:
Currently `timesheetsByEmployee` is un-idiomatic, since the return type is `Project[]`, not `Timesheet[]`. Perhaps `projectsByEmployeeTimesheets` would be more descriptive.